crates/ruff_python_formatter/src/other/arguments.rs

```diff
@@ -149,7 +149,10 @@ impl FormatNodeRule<Arguments> for FormatArguments {
 
             joiner.finish()?;
 
-            write!(f, [if_group_breaks(&text(","))])?;
+            // lambdas have no parentheses, so they also don't get a magic trailing comma
+            if self.parentheses != ArgumentsParentheses::SkipInsideLambda {
```

Comment:
Unrelated to this PR. Do we need to test if the magic trailing comma option is enabled?

Reply:
lambdas seem to have their own trailing comma rules that are different magic trailing comma

Reply:
That makes sense. But do we need to respect the magic trailing comma option for regular arguments?

Reply:
We do,
```python
def f(a,): pass
```
gets formatted as
```python
def f(
    a,
):
    pass
```

Reply:
But do we need to respect `context.options.magic_trailing_comma()`?

Reply:
sorry i misunderstand what you were aiming at, yes, for functions (but not lambdas) we need to respect the magic trailing comma setting and we previously didn't. I've added this now in this PR because separating these two aspects doesn't make sense